### PR TITLE
Add frame stroke option

### PR DESCRIPTION
**Pull request by Urgau**
This was proposed by `@nagisa` in https://github.com/jonhoo/inferno/issues/248.

The new option will be `--stroke-color` and default to `none` and accept any color.
It use a non-configurable stroke width of 1.

Fixes https://github.com/jonhoo/inferno/issues/248

### CODE REVIEW

File: src/flamegraph/mod.rs [hunk 1]
```diff
@@ -532,6 +539,12 @@ where
         ]),
     ))?;
 
+    let stroke_color = match opt.stroke_color {
+        StrokeColor::Color(c) => Some(c.to_string()),
+        StrokeColor::None => None,
+    };
```
Comment (jonhoo) [inline src/flamegraph/mod.rs:545]: Hmm, why is this necessary given you immediately do `as_deref` below?

File: src/bin/flamegraph.rs [hunk 2]
```diff
@@ -174,6 +174,14 @@ struct Opt {
     )]
     search_color: SearchColor,
 
+    /// Stroke color
```
Comment (jonhoo) [inline src/bin/flamegraph.rs:177]: We should document how this affects the design ("adds an outline to every frame") as well as the format here too I think. Probably also the stroke width.

File: src/bin/flamegraph.rs [hunk 3]
```diff
@@ -174,6 +174,14 @@ struct Opt {
...
+    #[clap(
+        long = "stroke-color",
+        default_value = defaults::STROKE_COLOR,
+        value_name = "STRING"
```
Comment (jonhoo) [inline src/bin/flamegraph.rs:181]: Why `STRING` and not `COLOR`?

File: src/flamegraph/mod.rs [hunk 4]
```diff
@@ -881,6 +902,13 @@ fn filled_rectangle<W: Write>(
             "fg:x" => &buffer[x_samples],
             "fg:w" => &buffer[width_samples]
         ));
+
+        if let Some(stroke_color) = stroke_color {
+            bytes_start.extend_attributes(args!(
+                "stroke" => stroke_color,
+                "stroke_width" => "1"
+            ));
+        }
```
Comment (jonhoo) [inline src/flamegraph/mod.rs:911]: It feels unfortunate to add this attribute to every single frame — can we instead just add it to the style sheet instead only when the option is set?

**Review by jonhoo — CHANGES_REQUESTED** [review]
Thanks for taking this on! I left a few comments.

File: src/bin/flamegraph.rs (hunk 3 above)
Comment (Urgau) [reply src/bin/flamegraph.rs:181]: Same as https://github.com/jonhoo/inferno/blob/c9c07d2f0248e646438b4cba9eae15ef587791d5/src/bin/flamegraph.rs#L170-L175 and also because "none" is not a color but a valid entry for the stroke color.

File: src/bin/flamegraph.rs (hunk 2 above)
Comment (Urgau) [reply src/bin/flamegraph.rs]: I'm not sure it's necessary to define the format here, it isn't done for the other options. As for the stroke width I also don't think we should mention it either, it doesn't add anything in my opinion.

Changed the text to `Adds an outline to every frame`.

File: src/flamegraph/mod.rs (hunk 4 above)
Comment (Urgau) [reply src/flamegraph/mod.rs]: Yes we can. I added them to `#frames > g > rect` like `@nagisa` was doing in their blog.

**Review by jonhoo — APPROVED** [review]
Nice, thanks!

**Comment by jonhoo** [comment]
Released in 0.11.7 :tada:
